### ISSUE TITLE
`deseq` results seemed weired when setting different `lfc_threshold` values 

### ISSUE THREAD

**Issue by jxshi** [hbc/bcbioRnaseq#194]
Hi @mjsteinbaugh ,

I recently noticed that when I change the value of `lfc_threshold` in the `Differential expression with DESeq2` template from 0 to 1, the results looked weired.
For example, when I used the following setting:
```
  alpha_threshold: 0.05
  lfc_threshold: 0
```
The `res <- results(dds, alpha = 0.05, lfcThreshold = 0); summary(res)` returned the following results:
```
out of 32634 with nonzero total read count
adjusted p-value < 0.05
LFC > 0 (up)       : 2426, 7.4%
LFC < 0 (down)     : 1698, 5.2%
outliers [1]       : 75, 0.23%
low counts [2]     : 10985, 34%
(mean count < 2)
```
But `res <- results(dds, alpha = 0.05, lfcThreshold = 1); summary(res)` returned very few differentially expressed genes:
```
out of 32634 with nonzero total read count
adjusted p-value < 0.05
LFC > 1.00 (up)    : 218, 0.67%
LFC < -1.00 (down) : 26, 0.08%
outliers [1]       : 75, 0.23%
low counts [2]     : 10985, 34%
(mean count < 2)
```
Then I manually checked the `all.csv` file exported from `res <- results(dds, alpha = 0.05, lfcThreshold = 0)`. Then I filtered the results by setting `padj < 0.05` and `log2FoldChange >=1` or `log2FoldChange <= -1`. It shows 1355 up-regulated genes and 479 down-regulated genes. This is quite different from the results when setting the following parameters:  `res <- results(dds, alpha = 0.05, lfcThreshold = 1); summary(res)` . 

This is the link to the [deseq.rds](https://1drv.ms/u/s!AuOKYFnYoxjlgqQkW-AmKFzbwOh-mg?e=oircAC) file and[bcb.rds](https://1drv.ms/u/s!AuOKYFnYoxjlgqQj7F6ctq6sPitzZA?e=f4DT4n) file, repectively. There are two groups: Model group with 5 replications and Control group with 5 replications.

Can you check for me, please?
Best,
Jianxiang

**Comment by mjsteinbaugh** [comment]
@jxshi That's correct regarding the usage of a non-zero `lfcThreshold` cutoff with `DESeq2::results`. Take a look at these resources, which explain the method:

- https://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#tests-of-log2-fold-change-above-or-below-a-threshold
- https://support.bioconductor.org/p/101504/ (very good explanation)
- https://support.bioconductor.org/p/104500/
- https://support.bioconductor.org/p/73774/
- https://support.bioconductor.org/p/9139890/
- "Hypothesis tests with thresholds on effect size" in the DESeq2 paper

So regarding this point:

> Then I manually checked the all.csv file exported from res <- results(dds, alpha = 0.05, lfcThreshold = 0). Then I filtered the results by setting padj < 0.05 and log2FoldChange >=1 or log2FoldChange <= -1. It shows 1355 up-regulated genes and 479 down-regulated genes. This is quite different from the results when setting the following parameters: res <- results(dds, alpha = 0.05, lfcThreshold = 1); summary(res) .

Basically the difference is due to you're applying a post-hoc cutoff of `lfcThreshold` when examining the results in `all.csv`, which differs from changing the statistical testing expectation of the LFC cutoff when calling `results`.

Best,
Mike

**Comment by jxshi** [comment]
Great explanation. Thanks!

Best,
Jianxiang
